R/process_data.R

```diff
@@ -62,6 +63,28 @@ process_pacta_results <- function(data, start_year, end_year, time_horizon,
   return(data_processed)
 }
 
+remove_companies_with_missing_exposures <- function(data) {
+  data <- data %>%
+    dplyr::group_by(
+      .data$investor_name, .data$portfolio_name, .data$equity_market,
+      .data$ald_sector, .data$technology, .data$scenario, .data$allocation,
+      .data$scenario_geography, .data$company_name
+    ) %>%
+    dplyr::arrange(.data$year) %>%
+    dplyr::mutate(
+      plan_carsten_missing = dplyr::if_else(
+        is.na(dplyr::last(.data$plan_carsten)),
```

Comment:
1. I do not understand why checking for nas on dplyr::last is the relevant criterion here. Can u please add a comment
2. I would suggest adding a message/warning/logline, you name it,  on which companies are removed

Reply:
1. re-wrote the function to hopefully clarify the logic
2. logging added